pkg/scheduler/plugins/proportion/capacity_policy/capacity_policy.go

```diff
@@ -83,3 +113,131 @@ func getRequiredQuota(tasksToAllocate []*pod_info.PodInfo) *podgroup_info.JobReq
 	}
 	return &quota
 }
+
+// getFirstPendingPod returns the first pod in a job that is in Pending status.
+// This is used to avoid duplicate quota checks for the same job.
+func getFirstPendingPod(job *podgroup_info.PodGroupInfo) *pod_info.PodInfo {
+	for _, pod := range job.PodInfos {
+		if pod.Status == pod_status.Pending {
+			return pod
+		}
+	}
+	return nil
+}
+
+// OnSessionOpen is called when a new scheduling session begins. It registers
+// the early quota checking function that prevents jobs from being considered
+// for scheduling if they would exceed their parent queues' quotas.
+func (cp *CapacityPolicy) OnSessionOpen(ssn *framework.Session) {
+	// Register early quota checks
+	ssn.AddPrePredicateFn(func(task *pod_info.PodInfo, job *podgroup_info.PodGroupInfo) error {
+		// Only check for the first pending pod to avoid duplicate checks
+		firstPending := getFirstPendingPod(job)
```

Comment:
Isn't the fact that the job needs scheduling (and then predicates are running on it) is not enough to understand that it has pending pods?